giddy/markov.py

```diff
@@ -1860,11 +1965,11 @@ def sojourn_time(p):
     pii = p.diagonal()
 
     if not (1 - pii).all():
-        print("Sojourn times are infinite for absorbing states!")
+        raise ValueError("Sojourn times are infinite for absorbing states!")
```

Comment:
Is there a way we can show the user that their matrix has absorbing states/explain what that means? I'm thinking something like:

```python
if not (1 - pii).all():
    absorbing_states, = (pii == 1).nonzero()
    raise ValueError('Sojurn times are infinite for absorbing states. '
                     'In this chain, states {} are absorbing states.'.format(list(absorbing_states)))
```

Reply:
This is very reasonable. Another way is to return `np.inf` for absorbing states without raising any value errors.

Reply:
This has been incorporated in https://github.com/pysal/giddy/pull/109/commits/b5792d53d7b9b8d58ffed22dc03f45579af79cf2